spring-integration-core/src/test/java/org/springframework/integration/aggregator/BarrierMessageHandlerTests.java

```diff
@@ -141,8 +142,8 @@ public void testReplyBeforeRequest() throws Exception {
 		handler.setOutputChannel(outputChannel);
 		handler.setBeanFactory(mock(BeanFactory.class));
 		handler.afterPropertiesSet();
-		Executors.newSingleThreadExecutor()
-				.execute(() -> handler.trigger(MessageBuilder.withPayload("bar").setCorrelationId("foo").build()));
+		ExecutorService exec = Executors.newSingleThreadExecutor();
```

Comment:
In the #2329 I went the way to expose a `SimpleAsyncTaskExecutor` as a class property do not instantiate per test and don't care about `shutdown`, but I think it really doesn't matter from test perspective.

So, merging...